### PR TITLE
Multiple notifications fixes

### PR DESCRIPTION
**Pull request by pfrazee**
- c1a2819fc79546a443ffcdc48e492486789bdf7b Removes a reset of the notifications feed cache that I'm fairly sure is the cause of the random loads of the feed while you're scrolling
- d78d5c00ceb97376b6aa9fdfa6d6242978e8a501 Small tweak of grouping behavior (reverting a change I tried earlier)
- b541f94f0ec6c61d7b06fa6ac2a3da25997dd725 Takes a new approach to resetting the notifs feed query on focus which avoids losing the user's scroll state 
- 05c39c92ee30d9c57870826d47847f6ed79199f8 Is a minor optimization, bringing back the use of the cached page from unread but only if the page is known to be fresh, avoiding the issue driving its removal in the first place (losing likes by the user)

### CODE REVIEW

File: src/state/queries/notifications/unread.tsx [hunk 1]
```diff
@@ -138,12 +138,12 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
       getCachedUnreadPage() {
         // return cached page if was for the current user
         // (protects against session changes serving data from the past session)
```
Comment (estrattonbailey) [inline src/state/queries/notifications/unread.tsx:140]: Outdated comment

File: src/lib/notifications/notifications.ts [hunk 2]
```diff
@@ -81,15 +81,14 @@ export function registerTokenChangeHandler(
 
 export function init(queryClient: QueryClient) {
   // handle notifications that are received, both in the foreground or background
+  // TODO: is this needed?
```
Comment (estrattonbailey) [inline src/lib/notifications/notifications.ts:84]: Doesn't seem like it. If someone taps on a push, we handle the invalidation and such then. No need to do anything ahead of time right now. Maybe keep this around just for debug logging?

File: src/state/queries/notifications/util.ts [hunk 3]
```diff
@@ -119,8 +119,7 @@ function groupNotifications(
           Math.abs(ts2 - ts) < MS_2DAY &&
           notif.reason === groupedNotif.notification.reason &&
           notif.reasonSubject === groupedNotif.notification.reasonSubject &&
-          notif.author.did !== groupedNotif.notification.author.did &&
-          notif.isRead === groupedNotif.notification.isRead
+          notif.author.did !== groupedNotif.notification.author.did
```
Comment (estrattonbailey) [inline src/state/queries/notifications/util.ts:122]: I THOUGHT I saw something like this, nice

File: src/view/screens/Notifications.tsx [hunk 4]
```diff
@@ -63,16 +64,26 @@ export function NotificationsScreen({}: Props) {
     }
   }, [scrollToTop, queryClient, unreadApi, hasNew])
 
+  const onFocusCheckLatest = React.useCallback(() => {
+    // on focus, check for latest, but only invalidate if the user
+    // isnt scrolled down to avoid moving content underneath them
+    unreadApi.checkUnread({invalidate: !isScrolledDown})
```
Comment (estrattonbailey) [inline src/view/screens/Notifications.tsx:70]: Yep that's nice

File: src/state/queries/notifications/types.ts [hunk 5]
```diff
@@ -28,7 +28,7 @@ export interface FeedPage {
 }
 
 export interface CachedFeedPage {
-  sessDid: string // used to invalidate on session changes
+  usableInFeed: boolean // if true, the cached page is recent enough to use as the response
```
Comment (estrattonbailey) [inline src/state/queries/notifications/types.ts:31]: Nit: could chuck this in a doc block and get it on hover where we use it

**Review by estrattonbailey — APPROVED** [review]
Works well from my testing

File: src/lib/notifications/notifications.ts (hunk 2 above)
Comment (pfrazee) [reply src/lib/notifications/notifications.ts:84]: yeah that's what I'm thinking

File: src/state/queries/notifications/types.ts (hunk 5 above)
Comment (pfrazee) [reply src/state/queries/notifications/types.ts]: Done

File: src/state/queries/notifications/unread.tsx (hunk 1 above)
Comment (pfrazee) [reply src/state/queries/notifications/unread.tsx]: Fixed

**Pull request merged**